mycore-base/src/main/java/org/mycore/frontend/basket/MCRBasketServlet.java

```diff
@@ -135,8 +135,13 @@ public void doGetPost(MCRServletJob job) throws Exception {
             getLayoutService().doLayout(req, res, new MCRJDOMContent(xml));
             return;
         }
-        if (referer != null && "referer".equals(redirect)) {
-            res.sendRedirect(res.encodeRedirectURL(referer.toExternalForm()));
+        if (referer != null) {
+            if ("referer".equals(redirect)) {
+                res.sendRedirect(res.encodeRedirectURL(referer.toExternalForm()));
+            }
+            else {
+                res.sendRedirect(res.encodeRedirectURL(redirect));
```

Comment:
If a referrer is given but the redirect parameter is empty. What happens?
If the redirect parameter is given but no referrer is available. What happens?